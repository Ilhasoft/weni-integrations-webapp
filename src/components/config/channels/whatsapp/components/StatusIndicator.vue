<template>
  <span :class="['status', `${status}`]" />
</template>

<script>
  export default {
    name: 'StatusIndicator',
    props: {
      status: {
        type: String,
        default: 'green',
        validator(value) {
          return ['green', 'yellow', 'red'].includes(value);
        },
      },
    },
  };
</script>

<style lang="scss" scoped>
  .status {
    height: 8px;
    border-radius: $unnnic-border-radius-pill;
    display: inline-block;

    width: 8px;
    min-width: 8px;
    align-self: center;

    &.green {
      background-color: $unnnic-color-feedback-green;
    }

    &.yellow {
      background-color: $unnnic-color-feedback-yellow;
    }

    &.red {
      background-color: $unnnic-color-feedback-red;
    }
  }
</style>
