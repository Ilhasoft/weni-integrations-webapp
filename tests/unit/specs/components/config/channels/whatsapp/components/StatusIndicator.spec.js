import { shallowMount, createLocalVue } from '@vue/test-utils';
import StatusIndicator from '@/components/config/channels/whatsapp/components/StatusIndicator.vue';

const localVue = createLocalVue();

describe('whatsapp/components/StatusIndicator.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(StatusIndicator, {
      localVue,
      propsData: {
        status: 'green',
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
