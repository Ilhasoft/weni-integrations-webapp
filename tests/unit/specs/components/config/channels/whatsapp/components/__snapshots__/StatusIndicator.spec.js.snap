// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`whatsapp/components/StatusIndicator.vue should be rendered properly 1`] = `<span class="status green"></span>`;
